### DISCUSSION TITLE
Compatibility issues with expansion units

### DISCUSSION THREAD

**Discussion by 1314Kissy** [007revad/Synology_HDD_db#207] [General]
Hello, I have encountered compatibility issues with HD6500 host NAS and expansion unit RX6022sas. Does RX6022sas exist in our script

**Comment by 007revad** [comment]
If you run these commands on the HD6500 do they list your RX6022sas?
```
syno_disk_data_collector record
file=$(ls /var/log/diskprediction | tail -n1)
grep -Eowi "([FRD]XD?[0-9]{3,4})(rp|ii|sas){0,2}" "/var/log/diskprediction/$file" 
```

**Comment by 1314Kissy** [comment]
![微信截图_20240109155529](https://github.com/007revad/Synology_HDD_db/assets/156056548/0c5d7b72-1b41-44be-9db6-b536cfcc498d)
I ran the script successfully and it also showed good green. After a while, it became unverified again

**Comment by 007revad** [comment]
Run the script with the -n option. That will stop DSM from updating the drive database.

**Comment by 1314Kissy** [comment]
Is it running like this  ./syno_hdd_db.sh -nf

**Comment by 007revad** [comment]
Yes.

But it is better to not use the -f option. Unless you need it.

**Comment by 1314Kissy** [comment]
![微信图片_20240109165312](https://github.com/007revad/Synology_HDD_db/assets/156056548/85ef0f3e-0d41-4512-8f00-582269bfea34)
Has I been stuck in this place all this time because I have too many hard drives? I have 60 hard drives, is it because there are too many hard drives

**Comment by 007revad** [comment]
Edit line 163 to change this:
```
    # set -x
```
to this:
```
    set -x
```
Then run it with the --debug option and let me know what it shows:
 ```
./syno_hdd_db.sh --debug
```

**Comment by 1314Kissy** [comment]
Thank you very much, it has been resolved.
